r/R/csv.R

```diff
@@ -692,7 +692,7 @@ write_csv_arrow <- function(x,
     x <- Table$create(x)
   }
 
-  assert_that(is_writable_table(x))
+  assert_that(is_writable_table(x) | inherits(x, "FileSystemDataset"))
```

Comment:
Might we also want a bare RecrodBatchReader class here too?

Reply:
Probably want this check moved inside `is_writeable_table()`, assuming that every other writer can take RecordBatchReaders (they should).

Reply:
They can't yet; there are some open tickets about this:

* [ARROW-14742: [C++] Allow ParquetWriter to take a RecordBatchReader as input](https://issues.apache.org/jira/browse/ARROW-14742)
* [ARROW-15405: [R] Allow all file writers to write from a RecordBatchReader](https://issues.apache.org/jira/browse/ARROW-15405)